doc/source/reference/functions/string_substring.rst

```diff
@@ -18,8 +20,97 @@ To enable this function, register ``functions/string`` plugin by following the c
 Syntax
 ------
 
+``string_substring`` requires two to four parameters.
+
+::
+
+  string_substring(target, nth[, options])
+  string_substring(target, nth, length[, options])
+
+``options`` uses the following format. All of key-value pairs are optional::
+
+  {
+    "default_value": default_value
+  }
+
 Usage
 -----
 
+Here are a schema definition and sample data to show usage.
+
+Sample schema:
+
+.. groonga-command
+.. include:: ../../example/reference/functions/string_substring/usage_setup_schema.log
+.. plugin_register functions/string
+.. table_create Memos TABLE_HASH_KEY ShortText
+
+Sample data:
+
+.. groonga-command
+.. include:: ../../example/reference/functions/string_substring/usage_setup_data.log
+.. load --table Memos
+.. [
+.. {"_key": "Groonga"}
+.. ]
+
+Here is a simple example for the extraction by position.
+
+.. groonga-command
+.. include:: ../../example/reference/functions/string_substring/usage_number.log
+.. select Memos --output_columns '_key, string_substring(_key, 2, 3)'
+
+In the following example, specifying the default value.
+
+.. groonga-command
+.. include:: ../../example/reference/functions/string_substring/usage_default.log
+.. select Memos --output_columns '_key, string_substring(_key, 50, 1, { "default_value" : "default" })'
+
+You can specify string literal instead of column.
+
+.. groonga-command
+.. include:: ../../example/reference/functions/string_substring/usage_string_literal.log
+.. select Memos --output_columns 'string_substring("Groonga", 2, 3)'
+
```

Comment:
We might as well write an example for specifying a negative value to the ``nth``.
Because I think this is the better explanation.